perfkitbenchmarker/pkb.py

```diff
@@ -195,35 +258,46 @@ def RunBenchmark(benchmark, collector, sequence_number, total_benchmarks):
         logging.exception('Prerequisite check failed for %s', benchmark_name)
         raise
 
-    start_time = time.time()
+    # Initialize timed interval objects.
+    end_to_end_interval = timed_interval.TimedInterval('End to End')
+    prepare_phase_interval = timed_interval.TimedInterval('Prepare Phase')
+    bm_prepare_interval = timed_interval.TimedInterval('Benchmark Prepare')
+    run_phase_interval = timed_interval.TimedInterval('Run Phase')
```

Comment:
For each of the phases, I added two sets of start/stop timestamps: one set for the entire phase, and one set for just the benchmark module's associated Prepare/Run/Cleanup function. "Benchmark Run" is timing just the call to the benchmark module's Run function. "Run Phase" includes the BEFORE_RUN_HOOKS and the collector.AddSamples call, which would also be called if running just pkb's run phase.